lib/sycamore/sycamore/transforms/text_extraction/ocr_models.py

```diff
@@ -0,0 +1,216 @@
+                return pages
+
+
+class EasyOcr(OcrModel):
+    @requires_modules("easyocr", extra="local-inference")
+    def __init__(self, lang_list=["en"]):
+        import easyocr
+
+        self.reader = easyocr.Reader(lang_list=lang_list)
+
+    def get_text(self, image: Image.Image) -> str:
+        image_bytes = BytesIO()
+        image.save(image_bytes, format="BMP")
+        raw_results = self.reader.readtext(image_bytes.getvalue())
+        out_list = []
+        for res in raw_results:
+            text = res[1]
+            out_list.append(text)
+        val = " ".join(out_list)
+        return val
+
+    def get_boxes_and_text(self, image: Image.Image) -> List[Dict[str, Any]]:
+        image_bytes = BytesIO()
+        image.save(image_bytes, format="BMP")
+        raw_results = self.reader.readtext(image_bytes.getvalue())
+
+        out: list[dict[str, Any]] = []
+        for res in raw_results:
+            raw_bbox = res[0]
+            text = res[1]
+            out.append(
+                {"bbox": BoundingBox(raw_bbox[0][0], raw_bbox[0][1], raw_bbox[2][0], raw_bbox[2][1]), "text": text}
+            )
+
+        return out
+
+    def __name__(self):
+        return "EasyOcr"
+
+
+class Tesseract(OcrModel):
+    @requires_modules("pytesseract", extra="local-inference")
+    def __init__(self):
+        import pytesseract
+
+        self.pytesseract = pytesseract
+
+    def get_text(self, image: Image.Image) -> str:
+        val = self.pytesseract.image_to_string(image)
+        return val
+
+    def get_boxes_and_text(self, image: Image.Image) -> List[Dict[str, Any]]:
+        output_list = []
+        base_dict = self.pytesseract.image_to_data(image, output_type=self.pytesseract.Output.DICT)
+        for value in zip(
+            base_dict["left"], base_dict["top"], base_dict["width"], base_dict["height"], base_dict["text"]
+        ):
+            if value[4] != "":
+                output_list.append(
+                    {
+                        "bbox": BoundingBox(value[0], value[1], value[0] + value[2], value[1] + value[3]),
+                        "text": value[4],
+                    }
+                )
+        return output_list
+
+    def __name__(self):
+        return "Tesseract"
+
+
+class LegacyOcr(OcrModel):
+    """Match existing behavior where we use tesseract for the main text and EasyOcr for tables."""
+
+    @requires_modules(["easyocr", "pytesseract"], extra="local-inference")
+    def __init__(self):
+        self.tesseract = Tesseract()
+        self.easy_ocr = EasyOcr()
+
+    def get_text(self, image: Image.Image) -> str:
+        return self.tesseract.get_text(image)
+
+    def get_boxes_and_text(self, image: Image.Image) -> List[Dict[str, Any]]:
+        return self.easy_ocr.get_boxes_and_text(image)
+
+    def __name__(self):
+        return "LegacyOcr"
+
+
+class PaddleOcr(OcrModel):
+    @requires_modules("paddleocr", extra="local-inference")
```

Comment:
Is there a way to declare a dependency for paddlepaddle here? Is the module name the same across CPU/GPU, or does it differ?

Reply:
It's `paddlepaddle` for CPU and `paddlepaddle-gpu` for GPU. I can make a comment noting this here.

Reply:
That's the name of the package, though, right? The module that you import is just called `paddle`? So I think you could still do `@requires_modules(["paddleocr", "paddle"], extra="local-inference")` . That will at least fail at runtime if you don't have either version of paddle installed. The error message won't tell you exactly what to install, but at least it will fail.

Reply:
Ah I see, yes that works.